plugins/secretstores/docker/docker_test.go

```diff
@@ -0,0 +1,178 @@
+package docker
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSampleConfig(t *testing.T) {
+	plugin := &Docker{}
+	require.NotEmpty(t, plugin.SampleConfig())
+}
+
+func TestInitFail(t *testing.T) {
+	tests := []struct {
+		name     string
+		plugin   *Docker
+		expected string
+	}{
+		{
+			name:     "invalid id",
+			plugin:   &Docker{},
+			expected: "id missing",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.plugin.Init()
+			require.ErrorContains(t, err, tt.expected)
+		})
+	}
+}
+
+func TestPathNonExistant(t *testing.T) {
+	plugin := &Docker{
+		ID:   "default_test",
+		Path: "non/existent/path",
+	}
+	err := plugin.Init()
+	require.Error(t, err)
+}
+
+func TestSetListGet(t *testing.T) {
+	secrets := map[string]string{
+		"secret-file-1": "I won't tell",
+		"secret_file_2": "secret",
+		"secretFile":    "foobar",
+	}
+
+	// Create a temporary directory we can use to store the secrets
+	testdir, err := os.MkdirTemp("", "docker-*")
+	require.NoError(t, err)
+	defer os.RemoveAll(testdir)
+
+	// Initialize the plugin
+	plugin := &Docker{
+		ID:   "test",
+		Path: testdir,
+	}
+	require.NoError(t, plugin.Init())
+
+	// Try to Store the secrets, which this plugin should not let
+	for k, v := range secrets {
+		require.ErrorContains(t, plugin.Set(k, v), "secret-store does not support creating secrets")
+	}
+
```

Comment:
Sorry but this does not make sense. You should check the error in an own testcase and remove it here.